### PR TITLE
Add support for specifying CID and guest ports to forward in go code

### PR DESCRIPTION
**Pull request by MagnusS**
- Add VSockPorts and VSockGuestCID to HyperKit struct. Guest CID is 3 by
  default.
- Add comma separated list of forwarded ports as -vsock-ports cli parameter in sample.

Signed-off-by: Magnus Skjegstad <magnus@skjegstad.com>

### CODE REVIEW

**Comment by rneugeba** [comment]
what's the use case for this? I seem to be able to connect to a vsock server in the VM just fine.

**Comment by MagnusS** [comment]
@rneugeba it's to create the sockets for the forwarded ports on startup, so you don't have to connect to them via the `connect` socket.

**Comment by MagnusS** [comment]
@rneugeba I've updated the commit message with a better description

File: go/hyperkit.go [hunk 1]
```diff
@@ -359,7 +381,12 @@ func (h *HyperKit) buildArgs(cmdline string) {
 		a = append(a, "-s", fmt.Sprintf("2:0,virtio-blk,%s", h.DiskImage))
 	}
 	if h.VSock {
-		a = append(a, "-s", fmt.Sprintf("3,virtio-sock,guest_cid=3,path=%s", h.StateDir))
+		l := fmt.Sprintf("3,virtio-sock,guest_cid=%d,path=%s", h.VSockGuestCID, h.StateDir)
+		if len(h.VSockPorts) > 0 {
+			l = fmt.Sprintf("%s,guest_forwards=%s", l, intArrayToString(h.VSockPorts, ";"))
+		}
+		a = append(a, "-s", l)
+		//fmt.Sprintf("3,virtio-sock,guest_cid=3,path=%s", h.StateDir))
```
Comment (rneugeba) [inline go/hyperkit.go]: shouldn't this be deleted?

File: go/sample/main.go [hunk 2]
```diff
@@ -7,11 +7,27 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
-	"log"
-
 	"github.com/docker/hyperkit/go"
+	"log"
+	"strconv"
+	"strings"
```
Comment (rneugeba) [inline go/sample/main.go]: these should go in the first import section and be ordered alphabetically

File: go/hyperkit.go (hunk 1 above)
Comment (MagnusS) [inline go/hyperkit.go]: oops, yes - fixed now

**Pull request merged**